### ISSUE TITLE
Unpack step failing occasionally with code 9

### ISSUE THREAD

**Issue by thoughtspacewebsites** [edjafarov/node-webkit-updater#94]
On occasion on OS X, I'm running into an issue during the unpacking portion of the update process that's resulting in the update failing. the file will completely download and as soon as I run the unpack function, I get an error. Here's the error:

{"killed":false,"code":9,"signal":null,"cmd":"/bin/sh -c unzip -xo \"/var/folders/64/xqb1tk3s0nbgwd_r8zb4sx200000gn/T/maple.1.2.2.zip\" >/dev/null"}

I'm not sure what's happening exactly, and I can't seem to pinpoint it to any specific actions. This has happened on two different machines so far both running OS X 10.11 El Capitan. Any insight greatly appreciated.

**Comment by adam-lynch** [comment]
Permissions? I'm not sure really to be honest.

**Comment by thoughtspacewebsites** [comment]
Nah, definitely not permissions. It's always completely random. I ended up "fixing" it by recursing my unpack function up to three times before throwing an error out to the user. Sometimes it will fail once or twice, but ever since I put in the recursive functionality, I haven't had a single visible error report, so that's good. It's not much of a fix, but it did take care of the immediate issue.

Like I said, it definitely isn't permission issues, especially considering sometimes it won't run the first time but then it will the second time in the exact same environment... Unfortunately the only error output I get is what I posted above. On this page:

http://linux.die.net/man/1/unzip

It looks like error code 9 is "the specified zipfiles were not found." Not sure if that helps at all. It's definitely a weird error. I'd like to say the download corrupted when the failures happen, but here's some other odd functionality: I don't redownload the files as part of my recursive function, the only part I recurse on is the unpack step. So the same ZIP that fails will succeed after two or three tries. It's really weird.
